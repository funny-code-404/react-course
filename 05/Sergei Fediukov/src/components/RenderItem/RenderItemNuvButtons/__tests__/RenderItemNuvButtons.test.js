import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import RenderItemNavButtons from '..'

configure({ adapter: new Adapter() });

describe('Should render correctly', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<RenderItemNavButtons />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('component has children button "go_back"', () => {
        expect(wrapper.find('.go_back')).toHaveLength(1)
    })
    it('component has children button "redirect"', () => {
        expect(wrapper.find('.redirect')).toHaveLength(1)
    })
})
