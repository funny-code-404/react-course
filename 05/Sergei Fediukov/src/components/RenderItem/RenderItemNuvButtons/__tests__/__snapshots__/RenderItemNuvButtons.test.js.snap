// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly to Match Snapshot 1`] = `
<div>
  <button
    className="go_back"
    onClick={[Function]}
  >
    go back
  </button>
  <button
    className="redirect"
    onClick={[Function]}
  >
    go home
  </button>
</div>
`;
